validators/bls.go

```diff
@@ -0,0 +1,136 @@
+package validators
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/0xPolygon/polygon-edge/helper/hex"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/fastrlp"
+)
+
+var (
+	ErrInvalidTypeAssert = errors.New("invalid type assert")
+)
+
+type BLSValidatorPublicKey []byte
+
+// String returns a public key in hex
+func (k BLSValidatorPublicKey) String() string {
+	return hex.EncodeToHex(k[:])
+}
+
+// MarshalText implements encoding.TextMarshaler
+func (k BLSValidatorPublicKey) MarshalText() ([]byte, error) {
+	return []byte(k.String()), nil
+}
+
+// UnmarshalText parses an BLS Public Key in hex
+func (k *BLSValidatorPublicKey) UnmarshalText(input []byte) error {
+	kk, err := hex.DecodeHex(string(input))
+	if err != nil {
+		return err
+	}
+
+	*k = kk
+
+	return nil
+}
+
+// BLSValidator is a validator using BLS signing algorithm
+type BLSValidator struct {
+	Address      types.Address
+	BLSPublicKey BLSValidatorPublicKey
+}
+
+// NewBLSValidator is a constructor of BLSValidator
+func NewBLSValidator(addr types.Address, blsPubkey []byte) *BLSValidator {
+	return &BLSValidator{
+		Address:      addr,
+		BLSPublicKey: BLSValidatorPublicKey(blsPubkey),
```

Comment:
nitpick, redudant type convertion. 
I think you can leave only
`BLSPublicKey: blsPubkey,`

Reply:
Fixed 2bd54a44bcb41d69e8bdbeec72957b5592ca86bd